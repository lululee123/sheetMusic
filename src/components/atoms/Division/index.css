.division {
	width: 100%;
	height: 1px;
	background-color: var(--mediumGray);

	&.vertical {
		height: 100%;
		width: 1px;
	}
}
