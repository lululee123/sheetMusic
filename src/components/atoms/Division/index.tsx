import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

interface DivisionProperty {
	className?: string;
	vertical?: boolean;
}

const Division: React.FC<DivisionProperty> = ({ className, vertical = false }) => (
	<div className={classnames(styles.division, className, { [styles.vertical]: vertical })} />
);

export default Division;
