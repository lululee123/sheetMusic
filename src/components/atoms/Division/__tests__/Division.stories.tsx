import React from 'react';
import { storiesOf } from '@storybook/react';

import Division from 'components/atoms/Division';

const stories = storiesOf('atoms/Division', module);

stories.add('__interactive', () => (
	<div style={{ marginTop: '20px' }}>
		<Division />
	</div>
));

stories.add('with vertical', () => (
	<div style={{ marginLeft: '20px', height: '100vh' }}>
		<Division vertical />
	</div>
));
