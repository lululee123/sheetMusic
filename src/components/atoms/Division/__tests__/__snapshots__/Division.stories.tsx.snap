// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    style="margin-top: 20px;"
  >
    <div
      class="division"
    />
  </div>
</DocumentFragment>
`;

exports[`with vertical 1`] = `
<DocumentFragment>
  <div
    style="margin-left: 20px; height: 100vh;"
  >
    <div
      class="division vertical"
    />
  </div>
</DocumentFragment>
`;
